src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/model/ContextObject.java

```diff
@@ -583,7 +601,9 @@ public FrameMarker getFrameMarker() {
 
     // The context represents primitive call which needs to be skipped when unwinding call stack.
     public boolean isPrimitiveContext() {
-        return !hasClosure() && getMethod().hasPrimitive() && getInstructionPointerForBytecodeLoop() <= CallPrimitiveNode.NUM_BYTECODES;
+        return !hasClosure() && getMethod().hasPrimitive() &&
+                        !getMethod().isUnwindMarked() && !getMethod().isExceptionHandlerMarked() &&
```

Comment:
Are these two additional checks really necessary?

Reply:
Well, as the comment in #ensure: says:
"This method is *not*
	 implemented as a primitive.  Primitive 198 always fails.  The VM uses prim
	 198 in a context's method as the mark for an ensure:/ifCurtailed: activation."
Similar for 199, these are just markers, not implemented as primitives, so they can never be "in" the primitive call, so this avoids the call to getInstructionPointerForBytecodeLoop() which I assume is more expensive, for several methods which are used commonly for flow control.
It can probably be made even cheaper